File: src/main/java/no/nav/aap/api/util/StreamUtil.java
package no.nav.aap.api.util;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

public final class StreamUtil {
    private StreamUtil() {
    }

    @SuppressWarnings("unchecked")
    public static <T> Stream<T> safeStream(T... elems) {
        return safeStream(Arrays.asList(elems));
    }

    public static <T> Stream<T> safeStream(List<T> list) {
        return Optional.ofNullable(list)
                .orElseGet(List::of)
                .stream();
    }
}

File: src/main/kotlin/no/nav/aap/api/søknad/util/StreamUtil.kt
package no.nav.aap.api.søknad.util

import java.util.*
import java.util.stream.Stream

object StreamUtil {
    fun <T> safeStream(vararg elems: T): Stream<T> {
        return safeStream(Arrays.asList(*elems))
    }

    @JvmStatic
    fun <T> safeStream(list: List<T>): Stream<T> {
        return Optional.ofNullable(list)
            .orElseGet { java.util.List.of() }
            .stream()
    }
}